### PR TITLE
allow ipv4 only

### PR DESCRIPTION
**Pull request by yanganto**
Related issue: https://github.com/kuutamolabs/kuutamocore/issues/720

### CODE REVIEW

File: nix/modules/toml-mapping.nix [hunk 1]
```diff
@@ -29,8 +29,8 @@ in
     kuutamo.network.ipv4.gateway = cfg.ipv4_gateway;
     kuutamo.network.ipv4.cidr = cfg.ipv4_cidr;
 
-    kuutamo.network.ipv6.address = cfg.ipv6_address;
-    kuutamo.network.ipv6.gateway = cfg.ipv6_gateway;
-    kuutamo.network.ipv6.cidr = cfg.ipv6_cidr;
+    kuutamo.network.ipv6.address = cfg.ipv6_address or null;
+    kuutamo.network.ipv6.gateway = cfg.ipv6_gateway or null;
+    kuutamo.network.ipv6.cidr = cfg.ipv6_cidr or null;
```
Comment (Mic92) [inline nix/modules/toml-mapping.nix:34]: null is not supported for this option at the moment.

```suggestion
    kuutamo.network.ipv6.cidr = cfg.ipv6_cidr or 128;
```

File: src/deploy/config.rs [hunk 2]
```diff
@@ -327,32 +327,40 @@ fn validate_host(
...
+        }
+
+        // FIXME: this is currently an unstable feature
+        //if ipv6_address.is_global() {
+        //    warn!("ipv6_address provided for hosts.{} is not a public ipv6 address: {}. This might not work with near mainnet", name, ipv6_address);
+        //}
+
+        (Some(ipv6_address), mask)
+    } else {
+        warn!("Please provide ipv6_address, which enable IPSec by default, and will provide a better authentication and confidentiality for you");
```
Comment (Mic92) [inline src/deploy/config.rs:361]: Where does this ipsec comes from?

File: nix/modules/toml-mapping.nix [hunk 3]
```diff
@@ -29,8 +29,8 @@ in
     kuutamo.network.ipv4.gateway = cfg.ipv4_gateway;
     kuutamo.network.ipv4.cidr = cfg.ipv4_cidr;
 
-    kuutamo.network.ipv6.address = cfg.ipv6_address;
-    kuutamo.network.ipv6.gateway = cfg.ipv6_gateway;
-    kuutamo.network.ipv6.cidr = cfg.ipv6_cidr;
+    kuutamo.network.ipv6.address = cfg.ipv6_address or null;
```
Comment (Mic92) [inline nix/modules/toml-mapping.nix:32]: I think we actually to forget to set this:

```suggestion
    kuutamo.kuutamod.publicAddress = cfg.ipv6_address or cfg.ipv4_address;
```

**Review by Mic92 — APPROVED** [review]

File: src/deploy/config.rs (hunk 2 above)
Comment (paulmattei) [reply src/deploy/config.rs]: What was the answer here...curious for my own information. It's nice if comments are resolved with explanation...

**Pull request merged**